lib/taskana-core-test/src/test/java/acceptance/TaskanaConfigurationTest.java

```diff
@@ -0,0 +1,70 @@
+package acceptance;
+
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+import java.lang.reflect.Field;
+import java.util.Collection;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.TestFactory;
+import org.junit.jupiter.api.function.ThrowingConsumer;
+
+import pro.taskana.TaskanaConfiguration;
+import pro.taskana.common.internal.util.ReflectionUtil;
+import pro.taskana.testapi.extensions.TestContainerExtension;
+
+class TaskanaConfigurationTest {
+
+  @TestFactory
+  Stream<DynamicTest> should_SaveUnmodifiableCollections() {
+    TaskanaConfiguration configuration =
+        new TaskanaConfiguration.Builder(
+                TestContainerExtension.createDataSourceForH2(), false, "TASKANA")
+            .build();
+
+    Stream<Field> fields =
+        ReflectionUtil.retrieveAllFields(TaskanaConfiguration.class).stream()
+            .filter(f -> Collection.class.isAssignableFrom(f.getType()));
+
+    ThrowingConsumer<Field> throwingConsumer =
```

Comment:
I prefer naming this `test` or something of that sort :) 
Applies to the next test case aswell